### PR TITLE
[10.0] [ADD] account_invoice_search_by_reference

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Search by vendor reference
==========================

This module adds the ability of searching by vendor reference when searching invoices from different views. This is useful for example when receiving supplier RMAs, where the user can search only by the internal invoice number.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

**Comment by LoisRForgeFlow** [comment]
Could you add some test cases? @nikul-serpentcs

**Comment by nikul-serpentcs** [comment]
Sure @lreficent I will do that.

**Comment by nikul-serpentcs** [comment]
@lreficent Could you please review Test cases.

File: account_invoice_search_by_reference/tests/test_account_invoice_search_by_reference.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo.tests.common import TransactionCase
+
+
+class TestAccountInvoiceSearchByReference(TransactionCase):
+    """
+        Tests for Account Invoice Merge.
```
Comment (LoisRForgeFlow) [inline account_invoice_search_by_reference/tests/test_account_invoice_search_by_reference.py]: you forgot to remove this

**Review by LoisRForgeFlow — APPROVED** [review]
Great, LGTM! Thanks - just a minor comment

File: account_invoice_search_by_reference/tests/test_account_invoice_search_by_reference.py (hunk 1 above)
Comment (nikul-serpentcs) [reply account_invoice_search_by_reference/tests/test_account_invoice_search_by_reference.py]: @lreficent  Removed line, Could you please review it.

File: account_invoice_search_by_reference/tests/test_account_invoice_search_by_reference.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            'reference': 'Test reference',
+            'invoice_line_ids': [(4, inv_line.id)],
+        })
+        return invoice
+
+    def test_account_invoice_method(self):
+        self.invoice1.action_invoice_open()
+        self.invoice1.write({'reference': 'Test'})
+        self.invoice1.name_search(name="INV", operator='ilike',
+                                  args=[('id', 'in', self.invoice1.ids)])
```
Comment (AaronHForgeFlow) [inline account_invoice_search_by_reference/tests/test_account_invoice_search_by_reference.py]: Can you add an  AssertEqual here?

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review Test cases.

**Review by AaronHForgeFlow — APPROVED** [review]
Functional + code :+1:  Thanks @nikul-serpentcs

File: account_invoice_search_by_reference/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
```
Comment (AaronHForgeFlow) [inline account_invoice_search_by_reference/__manifest__.py]: Just a minor thing but IMO the copyright should refer to authors or contributors who add new features to the module. Contributors that just migrate should go in the contributors section in the readme. Thanks.

**Review by houssine78 — APPROVED** [review]

**Comment by nikul-serpentcs** [comment]
@aheficent Improved copyright and squashed commits.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Thanks

**Comment by AaronHForgeFlow** [comment]
@nikul-serpentcs can you rebase this branch so diff is shown correct?

**Comment by AaronHForgeFlow** [comment]
@nikul-serpentcs can you please rebase again to see if this is still green?

**Comment by AaronHForgeFlow** [comment]
IMHO this is ok to merge. Is it possible to squash the merge commits?

**Comment by nikul-serpentcs** [comment]
@aheficent Actually the Merge Commits won't be possible to squash, as when you try to squash them it shows all the commits that are included in the PR instead of just the single Merge Commits!

**Comment by feketemihai** [comment]
@pedrobaeza To be merged...

**Review by feketemihai — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
If nobody disagree I'll merge this one

**Comment by feketemihai** [comment]
Yes.. It already has 3 approvals..

**Pull request merged**